elastic/kibana
# Add session cleanup audit logging

Resolves: #119490

## Summary

Add audit logging events for session cleanup

## Testing

1. Configure Kibana:

```yaml
xpack.security.audit.enabled: true
xpack.security.session.lifespan: "30s"
xpack.security.session.cleanupInterval: "10s"
```

2. Log in and immediately close the browser window
3. You should see the following event get logged in `logs/audit.log` after 40s

```json
{
  "@timestamp": "2022-01-04T18:56:29.541+00:00",
  "message": "Removing invalid or expired session for user [hash=eb28536c8ead72bf81a0a9226e38fc9bad81f5e07c2081bb801b2a5c8842924e]",
  "event": {
    "action": "session_cleanup",
    "category": [
      "authentication"
    ],
    "outcome": "unknown"
  },
  "user": {
    "hash": "eb28536c8ead72bf81a0a9226e38fc9bad81f5e07c2081bb801b2a5c8842924e"
  },
  "kibana": {
    "session_id": "o6SMKslvp1Ab9Edhm118PqM/zmz9+0m4mWLIc5tA4pQ=",
    "authentication_provider": "basic",
    "authentication_type": "basic"
  }
}
```
